### PR TITLE
A leaderboard!

### PR DESCRIPTION
**Pull request by bdecato**
Hey dudes,

I finally dove in, successfully set up a dev environment, and took a stab at the leaderboard (#7) . As implemented here, a user types `/leaderboard` and KarmaChameleon will read the entire karma file into a pandas data frame, separate it into user and non-user "things," and upload a plotly PNG image showing tables of karma sorted by net score for each category.

I preferred to output directly to markdown, but unfortunately Slack's send.message does not support nicely formatted text tables (see here).

One other thing I did that may have been a no-no was update the `requirements.txt` file to include everything in my `pip3 freeze` post virtualenv setup, plus the things I had to install for this pull request.

Let me know what you think and I'm happy to make useful modifications prior to merge.

Thanks!!

Ben

### CODE REVIEW

File: bot.py [hunk 1]
```diff
@@ -95,6 +98,42 @@ def chastise(self, inc: bool, channel_id: str):
         else:
             self._send_encouragement_message(channel_id)
 
+    def display_leaderboards(self, args: str, channel_id: str):
+        """
```
Comment (subecho) [inline bot.py:102]: You should be able to put this docstring on one line.

File: event_manager.py [hunk 2]
```diff
@@ -105,6 +106,18 @@ def listen():
     return None
 
 
+@app.route('/leaderboard', methods=['POST'])
+def slash_command():
```
Comment (subecho) [inline event_manager.py:110]: This should be renamed to `show_leaderboard` or something similar.

**Review by subecho — CHANGES_REQUESTED** [review]
Fantastic stuff! Just needs some small changes to be up to snuff :+1:

File: bot.py (hunk 1 above)
Comment (bdecato) [reply bot.py]: Addressed in 573ab863ac319d8900e15645b26f0f84c1145133!

File: event_manager.py (hunk 2 above)
Comment (bdecato) [reply event_manager.py:110]: Should this match the slash command exactly? The slash command is just /leaderboard currently.

Comment (subecho) [reply event_manager.py:110]: The route is fine, I just want the name of the method changed.

Comment (subecho) [reply event_manager.py:110]:
```suggestion
def show_leaderboard():
```

**Review by subecho — COMMENTED** [review]
Just needs to have the change applied for the method name and then we're good!

**Comment by bdecato** [comment]
FYI that the latest commit (9eb51d16d4e47d952de8335fddabd57cfdfeb073) fixes an unintended side-effect of the leaderboard as it was implemented, where every time a user uses the /leaderboard slash command, it will list out every person who has ever gotten a plus or minus *and also @ them.*

Obviously this is not great: if many people use the /leaderboard command, we will annoy our friends who rarely drop into N244. I should update the leaderboard to list usernames without explicitly @'ing.

Now we strip the @ mentions for each user in the leaderboard list, and also special-case handle @everyone, @here, and @channel.

**Review by bdecato — COMMENTED** [review]
I think all of these changes have been addressed!

**Review by subecho — APPROVED** [review]
Looks good! Ship it!

**Pull request merged**